distutils/sysconfig.py

```diff
@@ -265,7 +268,7 @@ def customize_compiler(compiler):  # noqa: C901
     Mainly needed on Unix, so we can plug in the information that
     varies across Unices and is stored in Python's Makefile.
     """
-    if compiler.compiler_type == "unix":
+    if compiler.compiler_type in ["unix", "cygwin", "mingw32"]:
```

Comment:
Maybe now that this value is selecting on multiple, it should be `isinstance(compiler, UnixCCompiler)`.

Reply:
`compiler.compiler_type` will never be `cygwin`, see https://github.com/pypa/distutils/issues/185, I'll better do a instance check instance.

Reply:
Changing to `isinstance` check fails the test suite. The `compiler` passed from the test suite is *not* a subclass of `UnixCCompiler` and fails this condition.